.devcontainer/devcontainer.json

```diff
@@ -1,18 +1,84 @@
 {
-	"name": "pwndbg",
-	"build": {
-		"context": "..",
-		"dockerfile": "../Dockerfile"
-	},
-	"runArgs": [
-		"--security-opt=no-new-privileges:true",
-		"--cap-drop=ALL",
-		"--cap-add=SYS_PTRACE"
-	],
-	"workspaceFolder": "/pwndbg",
-	"workspaceMount": "source=${localWorkspaceFolder},target=/pwndbg,type=bind,consistency=cached",
-	"settings": {},
-	"extensions": [
-		"ms-python.python"
-	]
-}
\ No newline at end of file
+    "name": "PwnDbg",
+    "build": {
+        "args": {},
+        "context": "..",
+        "dockerfile": "../Dockerfile"
+    },
+    "features": {},
+    "customizations": {
+        "vscode": {
+            "settings": {
+                "docker.languageserver.formatter.ignoreMultilineInstructions": true,
+                "editor.bracketPairColorization.enabled": true,
+                "editor.fontWeight": "bold",
+                "editor.formatOnSave": true,
+                "editor.guides.bracketPairs": "active",
+                "editor.guides.bracketPairsHorizontal": "active",
+                "editor.guides.highlightActiveBracketPair": true,
+                "editor.guides.highlightActiveIndentation": true,
+                "editor.guides.indentation": true,
+                "editor.renderWhitespace": "all",
+                "files.exclude": {
+                    "**/.git/**": true,
+                    "**/.mypy_cache/**": true,
+                    "**/__pycache__/**": true
+                },
+                "files.watcherExclude": {
+                    "**/.git/**": true,
+                    "**/.mypy_cache/**": true,
+                    "**/.venv/**": true,
+                    "**/__pycache__/**": true
+                },
+                "python.analysis.diagnosticMode": "workspace",
+                "python.analysis.typeCheckingMode": "strict",
+                "python.languageServer": "Pylance",
+                "python.linting.pylintUseMinimalCheckers": false,
+                "python.venvPath": "/venv",
+                "terminal.integrated.defaultProfile.linux": "zsh",
+                "[python]": {
+                    "editor.defaultFormatter": "ms-python.black-formatter",
+                    "editor.codeActionsOnSave": {
+                        "source.organizeImports": true
+                    },
+                    "editor.formatOnType": true
+                },
+                "[yaml]": {
+                    "editor.defaultFormatter": "redhat.vscode-yaml"
+                }
+            },
+            "extensions": [
```

Comment:
I'm rather against putting any extensions here unless we  really want a given extension and are sure it is official and really needed.

I mean, if someone wants to use an extension, they can do it on their own instead of us enforcing it on them.

This also helps us avoid users getting hacked by unofficial extensions becoming malicious and serving malicious updates.

Reply:
We discussed this on DMs on Discord and I accepted the extensions below. The only non-official one is ruff which seems ok.